### PR TITLE
Add InFlight Checks

### PR DESCRIPTION
**Pull request by jaronoff97**
- Full querying
- Shutdown and sleep

### CODE REVIEW

File: pkg/dependencies/customresource.go [hunk 1]
```diff
@@ -14,13 +14,13 @@ const (
...
 var (
-	CustomResourceClientInitializer = dependency[*apiextensionsclientset.Clientset]{
+	CustomResourceClientInitializer = dependency[apiextensionsclientset.Interface]{
 		dep:     NewCRDClient,
 		applier: checks.WithCustomResourceClient,
 	}
 )
 
-func NewCRDClient(ctx context.Context, http bool, token string, kubeconfig string) (*apiextensionsclientset.Clientset, *checks.Check) {
+func NewCRDClient(ctx context.Context, endpoint string, insecure bool, http bool, token string, kubeconfig string) (apiextensionsclientset.Interface, *checks.Check) {
```
Comment (codeboten) [inline pkg/dependencies/customresource.go:23]: nit: with this many repeated params, might be worth having a struct instead

File: pkg/dependencies/oteltrace.go [hunk 2]
```diff
@@ -2,10 +2,10 @@ package dependencies
 
 import (
 	"context"
-
-	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
 	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
 	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
```
Comment (codeboten) [inline pkg/dependencies/oteltrace.go]: nit: the grouping here looks incorrect

File: pkg/checks/runcrd/run.go [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+	results = append(results, checks.NewSuccessfulCheck(traceCheck, ""))
+	span.End()
+	results = append(results, checks.NewSuccessfulCheck(endTraceCheck, fmt.Sprintf("operation name: %s", operationName)))
+	err = c.tracer.Shutdown(ctx)
+	if err != nil && strings.Contains(err.Error(), "DeadlineExceeded") {
+		return append(results, checks.NewFailedCheck(traceFlush, deadlineExceeded, err))
+	} else if err != nil {
+		return append(results, checks.NewFailedCheck(traceFlush, badFlushMessage, err))
+	}
+	results = append(results, checks.NewSuccessfulCheck(traceFlush, "sent Trace"))
```
Comment (codeboten) [inline pkg/checks/runcrd/run.go]:
```suggestion
	results = append(results, checks.NewSuccessfulCheck(traceFlush, "sent trace"))
```

File: pkg/checks/runcrd/run.go [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+	if err != nil {
+		return append(results, checks.NewFailedCheck(queryMetricsCheck, "", err))
+	}
+	results = append(results, c.processMetrics(string(data))...)
+
+	err = c.client.Resource(colRes).Namespace(apiv1.NamespaceDefault).Delete(ctx, res.GetName(), metav1.DeleteOptions{})
+	if err != nil {
+		return append(results, checks.NewFailedCheck(deleteCollectorCheck, "", err))
+	}
+	return append(results, checks.NewSuccessfulCheck(deleteCollectorCheck, fmt.Sprintf("Deleted test collector %s", res.GetName())))
```
Comment (codeboten) [inline pkg/checks/runcrd/run.go]:
```suggestion
	return append(results, checks.NewSuccessfulCheck(deleteCollectorCheck, fmt.Sprintf("deleted test collector %s", res.GetName())))
```

**Review by codeboten — APPROVED** [review]
Just a couple of non-blocking suggestions

File: pkg/dependencies/customresource.go (hunk 1 above)
Comment (jaronoff97) [reply pkg/dependencies/customresource.go:23]: yep, going to refactor in the next PR.

**Pull request merged**